scripts/package

```diff
@@ -5,21 +5,52 @@ source $(dirname $0)/version
 
 cd $(dirname $0)/..
 
-PROJECT=`basename "$PWD"`
+PROJECT=$(basename "$PWD")
 
-if [ ! -x ./bin/longhorn ]; then
-    ./scripts/build
+command -v buildx >/dev/null && BUILD_CMD=(buildx) || BUILD_CMD=(docker buildx)
+
+# read configurable parameters
+REPO=${REPO:-longhornio}
+IMAGE_NAME=${IMAGE_NAME:-$PROJECT}
+TAG=${TAG:-''}
+OUTPUT_ARGS=${OUTPUT_ARGS:-'--load'}
+IS_SECURE=${IS_SECURE:-'false'}
+MACHINE=${MACHINE:-''}
+TARGET_PLATFORMS=${TARGET_PLATFORMS:-''}
+IID_FILE=${IID_FILE:-''}
+IID_FILE_FLAG=${IID_FILE_FLAG:-''}
+
+if [[ -z "$TAG" ]]; then
+    if API_VERSION=$(./bin/backing-image-manager version --client-only | jq ".clientVersion.backingImageManagerAPIVersion"); then
+      TAG="v${API_VERSION}_$(date -u +%Y%m%d)"
+    else
+      TAG="${VERSION}"
+    fi
 fi
 
-cp -r bin package/
+IMAGE="${REPO}/${IMAGE_NAME}:${TAG}"
 
-APIVERSION=`./bin/backing-image-manager version --client-only|jq ".clientVersion.backingImageManagerAPIVersion"`
-TAG="v${APIVERSION}_`date -u +%Y%m%d`"
-REPO=${REPO:-longhornio}
-IMAGE=${REPO}/${PROJECT}:${TAG}
+BUILDER_ARGS=()
+[[ ${MACHINE} ]] && BUILDER_ARGS+=('--builder' "${MACHINE}")
+
+IFS=' ' read -r -a IID_FILE_ARGS <<<"${IID_FILE_FLAG}"
```

Comment:
Why not name `$IID_FILE_FLAG` to `$IID_FILE_ARGS`

Reply:
`IID_FILE_FLAG` and/or `IID_FILE` would be externally assigned by [rancher/ecm-distro-tools/actions/publish-image](https://github.com/rancher/ecm-distro-tools/blob/master/actions/publish-image/action.yaml#L182).